src/FamilyService.Data/ChildRepository.cs

```diff
@@ -1,24 +1,63 @@
 ﻿using System;
+using System.Linq;
 using System.Threading.Tasks;
-using Microsoft.AspNetCore.Http;
+using System.Collections.Generic;
 using Microsoft.EntityFrameworkCore;
 using LT.DigitalOffice.FamilyService.Models.Db;
 using LT.DigitalOffice.FamilyService.Data.Provider;
 using LT.DigitalOffice.FamilyService.Data.Interfaces;
+using LT.DigitalOffice.FamilyService.Models.Dto.Requests.Child.Filters;
 
 namespace LT.DigitalOffice.FamilyService.Data
 {
   public class ChildRepository : IChildRepository
   {
     private readonly IDataProvider _provider;
-    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    private IQueryable<DbChild> CreateFindPredicates(
+      FindChildrenFilter filter,
+      IQueryable<DbChild> query,
+      List<Guid> departmentsUsers)
+    {
+      if (filter.Departments is not null 
+        && !departmentsUsers.Any())
+      {
+        return query.Where(dbChild => false);
+      }
+      
+      if(departmentsUsers is not null 
+        && departmentsUsers.Any())
+      {
+        query = query.Where(dbChild => departmentsUsers.Contains(dbChild.ParentUserId));
+      }
```

Comment:
may be lines 22-31 to
      if (filter.Departments.Any())
      {
        query = departmentsUsers.Any()
          ? query.Where(dbChild => departmentsUsers.Contains(dbChild.ParentUserId))
          : default;
      }

Reply:
The main idea of 22-31 lines was to avoid the following if statements due to waste of time because the query value is already empty.

Reply:
OK. replace check to Get
      if (filter is null || (filter.Departments.Any() && !departmentsUsers.Any())
      {
        return default;
      }